executor/runtime/docker/docker.go

```diff
@@ -1640,6 +1650,128 @@ func (r *DockerRuntime) Start(parentCtx context.Context, pod *v1.Pod) (string, *
 	return logDir, details, statusMessageChan, err
 }
 
+func (r *DockerRuntime) periodicallyReportContainerStatusMetrics(ctx context.Context) {
+	l := log.WithField("taskID", r.c.TaskID())
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
@sargun do you have any opinion on this technique here?
I don't want this go routine to ever crash titus-executor, but do you think this is overkill? (or has other unintended effects?)